[DOCS] Exception enhancements in 8.6 (#2748)
elastic/security-docs: docs/detections/add-exceptions.asciidoc

+
To exclude values from a
specific event in the sequence, update the rule's EQL statement. For example:
+
[source,eql]
----
`sequence
  [file where file.extension == "exe"
  and file.name != "app-name.exe"]
  [process where true
  and process.name != "process-name.exe"]`
----
==============

[float]
[[detection-rule-exceptions]]
=== Add exceptions to a rule

. Do one of the following:
+
--
* To add an exception from the rule details page:
.. Go to the rule details page of the rule to which you want to add an
exception (*Manage* -> *Rules* -> *_<Rule name>_*).
.. Scroll down the rule details page, select the *Rule exceptions* tab, then click *Add rule exception*.
+
[role="screenshot"]
image::images/rule-exception-tab.png[Detail of rule exceptions tab]

* To add an exception from the Alerts table:
.. Go to *Alerts*.
.. Scroll down to the Alerts table, go to the alert you want to create an exception for, click the *More Actions* menu (*...*), then select *Add rule exception*.

* To add an exception from the Rule Exceptions page:
.. Go to *Manage* -> *Rule Exceptions*.
.. Click *Create shared exception list* -> *Create exception item*. 
--

> . In the *Add Rule Exception* flyout, name the exception item and add conditions that define when the exception prevents alerts. When exception's query conditions are met (the query evaluates to `true`), rules do not generate alerts even when other rule criteria is met.

Author: ```suggestion
. In the *Add rule exception* flyout, name the exception item and add conditions that define when the exception prevents alerts. When exception's query conditions are met (the query evaluates to `true`), rules do not generate alerts even when other rule criteria is met.
```

+ 
In the example below, the exception prevents the rule from generating alerts when the `svchost.exe` process runs on hostname `siem-kibana`.
+
[role="screenshot"]
image::images/add-exception-ui.png[]

>   .. *Field*: Select a field to identify the event being filtered.

Author: ```suggestion
+
Add conditions that define when the exception prevents alerts:

  .. *Field*: Select a field to identify the event being filtered.
```

=======
    * `matches` | `does not match` — Allows you to use wildcards in *Value*, such as `C:\path\*\app.exe`. Available wildcards are `?` (match one character) and `*` (match zero or more characters). The selected *Field* data type must be {ref}/keyword.html#keyword-field-type[keyword], {ref}/text.html#text-field-type[text], or {ref}/keyword.html#wildcard-field-type[wildcard].
+
IMPORTANT: Using wildcards can impact performance. To create a more efficient exception using wildcards, use multiple conditions and make them as specific as possible. For example, adding conditions using `process.name` or `file.name` can help limit the scope of wildcard matching.

  .. *Value*: Enter the value associated with the *Field*. To enter multiple values (when using `is one of` or `is not one of`), enter each value, then press **Return**.

. Click *AND* or *OR* to create multiple conditions and define their relationships.

. Click *Add nested condition* to create conditions using nested fields. This is only required for
<<nested-field-list, these nested fields>>. For all other fields, nested conditions should not be used.

> . Choose to add the exception to the rule or a shared exception list. 

Reviewer: ```suggestion
. Choose to add the exception to a rule or a shared exception list. 
```
maybe?

> . Choose to add the exception to the rule or a shared exception list. 

Author: ```suggestion
. Choose to add the exception to a rule or a shared exception list. 
```

Author: Re-adding @benironside comment from [here](https://github.com/elastic/security-docs/pull/2748/files/add58aa1c620651778ee559e27fc12c4ed7f385f#r1038620714).

Author: Suggestion applied

+
You can add an exception to multiple rules if you are creating an exception item from the Rule Exceptions page. 
+ 
TIP: If a shared exception list doesn't exist, you can <<shared-exception-lists,create one>> from the Rule Exceptions page.

. (Optional) Enter a comment describing the exception.

. Select one of the following alert actions:

* *Close this alert*: Closes the alert when the exception is added. This option
is only available when adding exceptions from the Alerts table.
* *Close all alerts that match this exception and were generated by this rule*:
Closes all alerts that match the exception's conditions and were generated only by the current rule.
+

> . Click *Add Rule Exception*. 

Author: ```suggestion
. Click *Add rule exception*. 
```

[float]
[[endpoint-rule-exceptions]]
=== Add {elastic-endpoint} exceptions

> Like detection rule exceptions, you can add Endpoint agent exceptions either by editing the Endpoint Security rule or by adding them as an action on alerts generated by the Endpoint Security rule. {elastic-endpoint} alerts have the following fields:

Reviewer: ```suggestion
Like detection rule exceptions, you can add Endpoint agent exceptions either by editing the Endpoint Security rule or by adding them as actions on alerts generated by the Endpoint Security rule. {elastic-endpoint} alerts have the following fields:
```

* `kibana.alert.original_event.module determined:endpoint`
* `kibana.alert.original_event.kind:alert`

You can also add Endpoint exceptions to rules that are associated with {elastic-endpoint} rule exceptions. To associate rules, when creating or editing a rule, select the <<rule-ui-advanced-params, *{elastic-endpoint} exceptions*>> option.

Endpoint exceptions are added to

> both the detection rule *and* the {elastic-endpoint} agent on your hosts.

Author: Thanks for catching this! I didn't notice it before. I'm pretty sure "agent" isn't needed here, right @joepeeples ?

[IMPORTANT]
=============
Exceptions added to the Endpoint Security rule affect all alerts sent

> from the Endpoint agent. Be careful not to unintentionally prevent some Endpoint

Reviewer: ```suggestion
from the Endpoint agent. Be careful not to unintentionally prevent useful
```

Author: Hm, I'm split on this. I kind of want to keep "Endpoint" in the second sentence because I think the repetition helps emphasize the practice we're recommending. How about:

```suggestion
from the Endpoint agent. Be careful not to unintentionally prevent useful Endpoint
```

Author: Hm, I'm split on this. I kind of want to keep "Endpoint" in the second sentence because I think the repetition helps emphasize the practice we're recommending. How about:

```suggestion
from the Endpoint agent. Be careful not to unintentionally prevent useful Endpoint
```

Reviewer: Love it

alerts.

Additionally, to add an Endpoint exception to the Endpoint Security rule, there must be at least one Endpoint Security alert generated in the system. For non-production use, if no alerts exist, you can trigger a test alert using malware emulation techniques or tools such as the Anti Malware Testfile from the https://www.eicar.org/[European Institute for Computer Anti-Virus Research (EICAR)].
=============

[IMPORTANT]
=====
{ref}/binary.html[Binary fields] are not supported in detection rule exceptions.
=====

. Do one of the following:
+
--

* To add an Endpoint exception from the rule details page:

> .. Go to the rule details page (*Manage* -> *Rules*), and then search for and  select the Elastic *Endpoint Security* rule.

Reviewer: ```suggestion
.. Go to the rule details page (*Manage* -> *Rules*), and then search for and select the Elastic *Endpoint Security* rule.
```

> .. Go to the rule details page (*Manage* -> *Rules*), and then search for and  select the Elastic *Endpoint Security* rule.

Author: Re-adding Ben's comment from [here](https://github.com/elastic/security-docs/pull/2748/files/add58aa1c620651778ee559e27fc12c4ed7f385f#r1038622933).

```suggestion
.. Go to the rule details page (*Manage* -> *Rules*), and then search for and select the Elastic *Endpoint Security* rule.
```

.. Scroll down the rule details page, select the *Endpoint exceptions* tab, then click *Add endpoint exception*.

* To add an Endpoint exception from the Alerts table:
.. Go to *Alerts*.
.. Scroll down to the Alerts table, and from an {elastic-endpoint}
alert, click the *More actions* menu (*...*), then select *Add Endpoint exception*.

* To add an Endpoint exception from Rule Exceptions page:
.. Go to *Manage* -> *Rule Exceptions*.
.. Expand the Endpoint Security Exception List or click the list name to open the list's details page. Next, click *Add endpoint exception*. 
+

> NOTE: The Endpoint Security Exception List is automatically created. By default, it's associated with the Endpoint Security rule and with any rules that have the <<rule-ui-advanced-params, *{elastic-endpoint} exceptions*>> option selected.  

Reviewer 2: ```suggestion
NOTE: The Endpoint Security Exception List is automatically created. By default, it's associated with the Endpoint Security rule any rules with the <<rule-ui-advanced-params, *{elastic-endpoint} exceptions*>> option selected.  
```

Author: ```suggestion
NOTE: The Endpoint Security Exception List is automatically created. By default, it's associated with the Endpoint Security rule and any rules with the <<rule-ui-advanced-params, *{elastic-endpoint} exceptions*>> option selected.
```

Author: Re-added "and" because I think it was accidentally removed.

* *Close this alert*: Closes the alert when the exception is added. This option
is only available when adding exceptions from the Alerts table.
* *Close all alerts that match this exception and were generated by this rule*:
Closes all alerts that match the exception's conditions.

. Click *Add Endpoint Exception*. An exception is created for both the detection rule and the {elastic-endpoint}.

[float]
[[ex-nested-conditions]]
=== Exceptions with nested conditions

Some Endpoint objects contain nested fields, and the only way to ensure you are
excluding the correct fields is with nested conditions. One example is the
`process.Ext` object:

[source, json]
--------------------------------------------------
{
  "ancestry": [],
  "code_signature": {
    "trusted": true,
    "subject_name": "LFC",
    "exists": true,
    "status": "trusted"
  },
  "user": "WDAGUtilityAccount",
  "token": {
    "elevation": true,
    "integrity_level_name": "high",
    "domain": "27FB305D-3838-4",
    "user": "WDAGUtilityAccount",
    "elevation_type": "default",
    "sid": "S-1-5-21-2047949552-857980807-821054962-504"
  }
}
--------------------------------------------------

> TIP: `code_signature.subject_name` refers to the process signature not the

Reviewer: I know this tip was here before, but I'm trying to figure out how it connects to anything else on the page? No other mentions of `code_signature.subject_name`. Maybe cut it?

Author: I think this is mainly a tip about the example in lines 172-192. I'm ok with removing it though if it's not all that relevant to exceptions in general. :)

* *Close this alert*: Closes the alert when the exception is added. This option
is only available when adding exceptions from the Alerts table.
* *Close all alerts that match this exception and were generated by this rule*:
Closes all alerts that match the exception's conditions.

. Click *Add Endpoint Exception*. An exception is created for both the detection rule and the {elastic-endpoint}.

[float]
[[ex-nested-conditions]]
=== Exceptions with nested conditions

Some Endpoint objects contain nested fields, and the only way to ensure you are
excluding the correct fields is with nested conditions. One example is the
`process.Ext` object:

[source, json]
--------------------------------------------------
{
  "ancestry": [],
  "code_signature": {
    "trusted": true,
    "subject_name": "LFC",
    "exists": true,
    "status": "trusted"
  },
  "user": "WDAGUtilityAccount",
  "token": {
    "elevation": true,
    "integrity_level_name": "high",
    "domain": "27FB305D-3838-4",
    "user": "WDAGUtilityAccount",
    "elevation_type": "default",
    "sid": "S-1-5-21-2047949552-857980807-821054962-504"
  }
}
--------------------------------------------------

> process name.

Author: ```suggestion

```

* *Close this alert*: Closes the alert when the exception is added. This option
is only available when adding exceptions from the Alerts table.
* *Close all alerts that match this exception and were generated by this rule*:
Closes all alerts that match the exception's conditions.

. Click *Add Endpoint Exception*. An exception is created for both the detection rule and the {elastic-endpoint}.

[float]
[[ex-nested-conditions]]
=== Exceptions with nested conditions

Some Endpoint objects contain nested fields, and the only way to ensure you are
excluding the correct fields is with nested conditions. One example is the
`process.Ext` object:

[source, json]
--------------------------------------------------
{
  "ancestry": [],
  "code_signature": {
    "trusted": true,
    "subject_name": "LFC",
    "exists": true,
    "status": "trusted"
  },
  "user": "WDAGUtilityAccount",
  "token": {
    "elevation": true,
    "integrity_level_name": "high",
    "domain": "27FB305D-3838-4",
    "user": "WDAGUtilityAccount",
    "elevation_type": "default",
    "sid": "S-1-5-21-2047949552-857980807-821054962-504"
  }
}
--------------------------------------------------

> TIP: `code_signature.subject_name` refers to the process signature, not the

Author: ```suggestion

```

Author: Removing as per suggestion [here](https://github.com/elastic/security-docs/pull/2748#discussion_r1039121452).
